### PR TITLE
Guiding the user to change their message if a file is sent.

### PR DESCRIPTION
**Pull request by makennahardy**
Rubber duck tracks if a message has a file attachment from the user. If so, rubber duck asks the user to rephrase their question and paste only relevant sections of the file. It is important to note that If the user asks a question _and_ pastes a file, the entire question and entire file are disregarded and not stored in the message history. This is why rubber duck prompts the user to rephrase their question as well as paste only part of the file.

### CODE REVIEW

**Comment by makennahardy** [comment]
*Question*

- On line 120 of rubber_duck.py, I pass in the hard-coded rubber duck response (when there is a file pasted from the user) to self._metrics_handler.record_message(). I do not, however, pass anything to self._metrics_handler.record_usage() because no tokens are used in the exchange. In regard to metrics handling, is this how information should be tracked when the user attaches a file?

File: discord_bot.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def as_message(message: discord.Message) -> Message:
         author_name=message.author.name,
         author_mention=message.author.mention,
         message_id=message.id,
-        content=message.content
+        content=message.content,
+        is_file=len(message.attachments) > 0
```
Comment (byubean) [inline discord_bot.py:70]: I assume the file is some kind of URL or resource ID. We should pass that through in the `Message` so we have the option of using it in the future.

File: rubber_duck.py [hunk 2]
```diff
@@ -28,6 +28,7 @@ class Message(TypedDict):
     author_mention: str
     message_id: int
     content: str
+    is_file: bool
```
Comment (byubean) [inline rubber_duck.py:31]: So instead of `is_file`, we want `file: bool | None`

File: rubber_duck.py [hunk 3]
```diff
@@ -98,7 +99,12 @@ async def have_conversation(self, thread_id: int, engine: str, prompt: str, init
                     await self._send_message(thread_id, '*This conversation has been closed.*')
                     return
 
-                message_history.append(GPTMessage(role='user', content=message['content']))
+                if message['is_file'] == True:
+                    appended_content = "*The user attached a file*"
+                else:
+                    appended_content = message['content']
+
+                message_history.append(GPTMessage(role='user', content=appended_content))
```
Comment (byubean) [inline rubber_duck.py:107]: Instead, just insert a line before the original line 101 that:
```python
if message['file'] is not None:
    self._send_message(thread_id, "I'm sorry, I can't read file attachments. Please resend your message with the relevant parts of your file included in the message")
    continue
```

**Review by byubean — CHANGES_REQUESTED** [review]
See comments

**Pull request merged**